src/emscripten-optimizer/simple_ast.h

```diff
@@ -920,22 +920,22 @@ struct JSPrinter {
         char *test = end;
         // remove zeros, and also doubles can use at most 24 digits, we can truncate any extras even if not zero
         while ((*test == '0' || test - buffer > 24) && test > buffer) test--;
-        int num = end - test;
+        size_t num = end - test;
         if (num >= 3) {
           test++;
           test[0] = 'e';
           if (num < 10) {
-            test[1] = '0' + num;
+            test[1] = (char)('0' + num);
```

Comment:
Yet another thing I learned today.

    On Friday, May 13, 2016 10:38 PM, Derek Schuff <notifications@github.com> wrote:
 

 In src/emscripten-optimizer/simple_ast.h:>          if (num >= 3) {
>            test++;
>            test[0] = 'e';
>            if (num < 10) {
> -            test[1] = '0' + num;
> +            test[1] = (char)('0' + num);
It is in C; It's actually not in C++ (http://en.cppreference.com/w/cpp/language/character_literal)—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub